tests/src/runtimeApi/memory/hipMemcpy.cpp

```diff
@@ -291,12 +318,20 @@ void memcpytest2_sizes(size_t maxElem = 0) {
     int deviceId;
     HIPCHECK(hipGetDevice(&deviceId));
 
-    size_t free, total;
+    size_t free, total, freeCPU, totalCPU;
     HIPCHECK(hipMemGetInfo(&free, &total));
+    memcpytest2_get_host_memory(freeCPU, totalCPU);
 
     if (maxElem == 0) {
-        maxElem = free / sizeof(T) / 8;
+        // Use lesser maxElem if not enough host memory available
+        size_t maxElemGPU = free / sizeof(T) / 8;
+        size_t maxElemCPU = freeCPU / sizeof(T) / 8;
```

Comment:
Probably  use maxElemCPU = freeCPU / sizeof(T) / 4 here. Then total host memory used is freeCPU/2 since there are two host arrays allocated. Then it should still be safe.

Reply:
Hi Sam, it seems that each call to HostMemory<T> will allocate 2 arrays. So in this test, we are allocating 4 host memory array, 2 pinned and 2 not. So if we give freecpu/4, won't it occupy most of the cpu mem?

Reply:
You are right. So let's use maxElemCPU = freeCPU / sizeof(T) / 8